cartography/data/indexes.cypher

```diff
@@ -451,3 +451,10 @@ CREATE INDEX IF NOT EXISTS FOR (n:KubernetesSecret) ON (n.lastupdated);
 CREATE INDEX IF NOT EXISTS FOR (n:KubernetesService) ON (n.id);
 CREATE INDEX IF NOT EXISTS FOR (n:KubernetesService) ON (n.name);
 CREATE INDEX IF NOT EXISTS FOR (n:KubernetesService) ON (n.lastupdated);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngress) ON (n.id);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngress) ON (n.name);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngress) ON (n.lastupdated);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngressRule) ON (n.ingress_uid, n.host);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngressRule) ON (n.lastupdated);
+CREATE INDEX IF NOT EXISTS FOR (n:KubernetesIngressRuleHttpPath) ON (n.ingress_uid, n.host, n.path, n.path_type);
```

Comment:
This would be our first time using composite indexes. There could be some performance implications.
https://neo4j.com/docs/cypher-manual/4.1/administration/indexes-for-search-performance/#administration-indexes-single-vs-composite-index

Have you noticed any issues when you do the single indexes?

Reply:
No, it went really fine. I have been careful with the order to be future-usages-proof (as for now, we use all the keys when we use this index, as it is a kind of "id" of the rule or the path).

Reply:
I could do that, but:
- compound indexes are meant exactly for that
- such `id` index would not then be suitable for lookups on just the ingress_uid, for example

Reply:
Please do that. 
Why do you think it would not be appropriate?

Reply:
What I'm saying is: compound indexes are done for that kind of use cases. They have the advantages of

- being straightforward (you just use them with the API)
- being usable with only some keys (you don't have to create a regexp to look for all rules within a given ingress, for example)

I see no advantage of creating a pseudo field with a string concatenation of different values. That's exactly where compound indexes are needed.

Reply:
The advantage is that we will have an `id` for these new nodes, just like we have for every other node in our index. The n=Neo4j docs also advise that users come up with their own Node ids, rather than leaving the ids unassigned.

You may keep the compound indexes, but you must have an appropriate `id` for the new nodes.